### PR TITLE
Redesign filters topic page

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: public/locales/he/common.json [hunk 1]
```diff
@@ -101,6 +101,18 @@
...
+    "aside-filter": {
+      "filters-title": "סנן את מה שמעניין אותך:",
+      "sources-filter-multi": "מקובלים",
+      "topics-filter-multi": "נושאים",
+      "content-type-filter": "מדור",
+      "language-filter": "שפה",
+      "date-filter": "תאריך",
+      "lessons": "שיעורים והרצאות",
+      "events": "כנסים ואירועים",
+      "search-input-topic": "search topic",
```
Comment (wissog) [inline public/locales/he/common.json:114]: we need to translate this

File: public/locales/ru/common.json [hunk 2]
```diff
@@ -102,6 +102,18 @@
...
+      "filters-title": "Фильтровать по интересам:",
+      "sources-filter-multi": "Каббалисты",
+      "topics-filter-multi": "Темы",
+      "content-type-filter": "Раздел",
+      "language-filter": "Язык",
+      "date-filter": "Дата",
+      "lessons": "Уроки и лекции",
+      "events": "Конгрессы, события",
+      "search-input-topic": "search topic",
+      "search-input-source": "search source"
```
Comment (wissog) [inline public/locales/ru/common.json:116]: hebrew and russian we may do ourselves

File: src/components/shared/ContentItem/ListTemplate.js [hunk 3]
```diff
@@ -73,10 +90,33 @@ const ListTemplate = ({
...
       <div className={clsx('cu_item_info', { [dir]: true, 'with_actions': !!children })}>
+        {
+          withCUInfo && (
+            <>
+              <Ref innerRef={cuInfoRef}>
+                <Header
+                  as={size === 'big' || isMobileDevice ? 'h5' : 'h3'}
+                  className={clsx('cu_item_name', { 'show_part': !cuInfoShowAll })}
+                >
+                  {(unit && unit.name) || (source && source.name) || (tag && tag.label)}
```
Comment (wissog) [inline src/components/shared/ContentItem/ListTemplate.js:101]: this can be shortened like: unit?.name, source?.name, tag?.label

**Review by wissog — DISMISSED** [review]

**Review by bbfsdev — COMMENTED** [review]
On wide monitor there are some spaces:
image


The following should be removed (show more / less):
image
You can recreate this by clicking "hanucha" filter.

File: src/components/Filters/components/Date/helper.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  LAST_30_DAYS
+];
+
+export const presetToRange = {
+  [TODAY]: () => {
+    const nToday = today().toDate();
+    return ({ from: nToday, to: nToday });
+  },
+  [LAST_7_DAYS]: () => ({
+    from: today().subtract(6, 'days').toDate(),
```
Comment (bbfsdev) [inline src/components/Filters/components/Date/helper.js:31]: Why subtract 6 for 7 days?

File: src/components/Filters/components/Date/helper.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    to: today().toDate()
+  }),
+  [LAST_30_DAYS]: () => ({
+    from: today().subtract(29, 'days').toDate(),
```
Comment (bbfsdev) [inline src/components/Filters/components/Date/helper.js:35]: Same here, why 29 for 30?

**Review by bbfsdev — COMMENTED** [review]
The checkboxes are not disabled while things are loading:
https://drive.google.com/file/d/14HUOtNDjsFqRSjw15-gEiwcmAAIhH-eA/view

Not sure what is the right solution.

File: src/components/FiltersAside/ContentTypeFilter/ContentType.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  const lStats   = useSelector(state => UNIT_LESSONS_TYPE.reduce(
+    (acc, x) => acc + filtersAside.getStats(state.filtersAside, namespace, FN_CONTENT_TYPE, x), 0)
+  );
+  const eStats   = useSelector(state => UNIT_EVENTS_TYPE.reduce(
+    (acc, x) => acc + filtersAside.getStats(state.filtersAside, namespace, FN_CONTENT_TYPE, x), 0)
+  );
+
+  const dispatch = useDispatch();
+
+  if (!(items?.length > 0)) return null;
```
Comment (bbfsdev) [inline src/components/FiltersAside/ContentTypeFilter/ContentType.js:36]: if (items?.length === 0) return null;

File: src/components/FiltersAside/DateFilter.js [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+    setTo(r?.to);
+    setPreset(r ? p : null);
+    setShowRange(p === CUSTOM_RANGE);
+    setShowDay(p === CUSTOM_DAY);
+  }, [selected]);
+
+  const updateFilter = (_preset, _from, _to) => {
+    const r = buildRange(_preset, _from, _to);
+    dispatch(actions.setFilterValue(namespace, FN_DATE_FILTER, r));
+
```
Comment (bbfsdev) [inline src/components/FiltersAside/DateFilter.js:51]: Remove empty line.

File: src/components/FiltersAside/DateFilter.js [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+  const buildRange = (_preset, _from = from, _to = to) => {
+    if (!_preset) return null;
+    // calculate range with regard to the date preset
+    if (_preset === CUSTOM_RANGE || _preset === CUSTOM_DAY) {
+      return { from: _from, to: _to };
+    }
+
+    return (presetToRange[_preset] || presetToRange[TODAY])();
+
```
Comment (bbfsdev) [inline src/components/FiltersAside/DateFilter.js:62]: Remove empty line.

File: src/components/FiltersAside/DateFilter.js [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+        </List.Item>
+      </Accordion>
+    </Segment.Group>
+  );
+  return (
+    <FilterHeader
+      filterName={FN_DATE_FILTER}
+      children={renderContent()}
+    />
+
```
Comment (bbfsdev) [inline src/components/FiltersAside/DateFilter.js:182]: Remove empty line.

File: src/sagas/tags.js [hunk 10]
```diff
@@ -4,23 +4,39 @@ import Api from '../helpers/Api';
...
+
+    const cuIDs    = items.map(x => x.content_unit_id);
+    const labelIDs = items.map(x => x.label_id).filter(x => !!x);
+    const list     = items.map(({ content_unit_id, label_id, is_text }) => ({
+      cuID: content_unit_id,
+      lID: label_id,
+      isText: is_text
+    }));
+
+    yield put(mdb.fetchUnitsByIDs({ id: cuIDs }));
```
Comment (bbfsdev) [inline src/sagas/tags.js:36]: Can you do all those in parallel?
Why one after another?
Are those done in parallel?

**Review by bbfsdev — APPROVED** [review]

File: src/components/Filters/components/Date/helper.js (hunk 4 above)
Comment (davgur) [reply src/components/Filters/components/Date/helper.js:31]: - its copy/paste from usual kmedia date filter
- additional day is today

File: public/locales/he/common.json (hunk 1 above)
Comment (davgur) [reply public/locales/he/common.json]: removed

File: public/locales/ru/common.json (hunk 2 above)
Comment (davgur) [reply public/locales/ru/common.json]: removed

File: src/components/shared/ContentItem/ListTemplate.js (hunk 3 above)
Comment (davgur) [reply src/components/shared/ContentItem/ListTemplate.js]: replaced

File: src/sagas/tags.js (hunk 10 above)
Comment (davgur) [reply src/sagas/tags.js]: done

File: src/components/FiltersAside/DateFilter.js (hunk 9 above)
Comment (davgur) [reply src/components/FiltersAside/DateFilter.js:182]: done

File: src/components/FiltersAside/DateFilter.js (hunk 8 above)
Comment (davgur) [reply src/components/FiltersAside/DateFilter.js:62]: done

File: src/components/FiltersAside/DateFilter.js (hunk 7 above)
Comment (davgur) [reply src/components/FiltersAside/DateFilter.js:51]: done

**Review by bbfsdev — APPROVED** [review]

**Pull request merged**